### PR TITLE
FIX: add additional condition to check invalid type(#28277)

### LINKED ISSUE #28277: `series.str.cat(series.str)` is concatenating only the largest string
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd

arr = ["AbC", "de", "FGHI", "j", "kLLLm"]

ps = pd.Series(arr)
expect = ps.str.cat(others=ps.str)
print(expect)
Out[16]:
0           NaN
1           NaN
2           NaN
3           NaN
4    kLLLmkLLLm
dtype: object
```

#### Problem description
`series.str.cat(series.str)` is concatenating only the largest string in the series but it should concatenate all the strings element wise.

#### Expected Output
```python
Out[18]:
0        AbCAbC
1          dede
2      FGHIFGHI
3            jj
4    kLLLmkLLLm
dtype: object
```

#### Output of ``pd.show_versions()``

<details>

In [2]: pandas.show_versions()

INSTALLED VERSIONS
------------------
commit: None
python: 3.7.3.final.0
python-bits: 64
OS: Linux
OS-release: 3.10.0-862.14.4.el7.x86_64
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: None
LOCALE: en_US.UTF-8

pandas: 0.24.2
pytest: 5.1.2
pip: 19.2.3
setuptools: 41.2.0
Cython: 0.29.13
numpy: 1.17.1
scipy: None
pyarrow: 0.14.1
xarray: None
IPython: 7.8.0
sphinx: 2.2.0
patsy: None
dateutil: 2.8.0
pytz: 2019.2
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml.etree: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10.1
s3fs: 0.3.4
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None

</details>

### PR DESCRIPTION
**Pull request by boomsquared**
- [ ] closes #28277
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by WillAyd** [comment]
Thanks for the PR! Can you add a test for the behavior you are trying to solve? Tests are typically the most critical part of any PR

**Comment by bangsquared** [comment]
@WillAyd Thanks for the advice, will certainly do.

File: pandas/core/strings.py [hunk 1]
```diff
@@ -2107,7 +2107,9 @@ def _get_series_list(self, others):
         elif isinstance(others, np.ndarray) and others.ndim == 2:
             others = DataFrame(others, index=idx)
             return [others[x] for x in others]
-        elif is_list_like(others, allow_sets=False):
+        elif is_list_like(others, allow_sets=False) and not isinstance(
```
Comment (WillAyd) [inline pandas/core/strings.py]: Not sure about this. I *think* the real issue is that `is_list_like` returns True for `s.str` when it shouldn't, though I do see that is also technically Iterable so maybe this should be allowed and just fixed

**Review by WillAyd — CHANGES_REQUESTED** [review]
Thanks again for the PR! Great investigation into root cause. I'm not sure about the proposed fixed just yet but cc @TomAugspurger if he has thoughts as well

Can you also add a whatsnew note for v1.0.0?

**Comment by TomAugspurger** [comment]
I'm not sold on the value of `Series.str.__iter__`, but https://github.com/pandas-dev/pandas/issues/3638 and https://github.com/pandas-dev/pandas/pull/3645 are interesting history. I'm not sure about the value of `Series.str.__iter__`, especially when it causes surprising behavior like this.

**Comment by WillAyd** [comment]
Hmm that's very surprising. Seems like it yields "columnar character by character" access across all elements?

Maybe we should just deprecate that behavior then? Having a hard time seeing where that is useful as an iterator

**Comment by boomsquared** [comment]
@WillAyd  The StringMethods.get calls str_get which str_get "Extract element from lists, tuples, or strings in each element in the  Series/Index". Should I fix the StringMethods.get by not calling str_get?

**Comment by WillAyd** [comment]
I think the best approach would be to deprecate the `__iter__` method of `Series.str` as noted above and remove in the future - is that something you'd be interested in trying?

**Comment by boomsquared** [comment]
@WillAyd Is there something more to depreciating the function than just removing it from the code?

**Comment by WillAyd** [comment]
Yea we would want to throw a warning for the time being and add some doc notes / sphinx directives for the deprecation. There's a Deprecate label where you can see past PRs, but citing a specific example I think #26744 is one PR that's relatively clean to emulate

**Pull request closed without merging**